invenio_records_resources/services/files/schema.py

```diff
@@ -85,7 +85,20 @@ def fields_from_file_obj(self, data, **kwargs):
         if data.file:
             # mandatory fields
             data["storage_class"] = data.file.storage_class
-            data["uri"] = data.file.uri  # TODO: only serialize for external?
+            data["uri"] = data.file.uri
+
+            # If Local -> remove uri as it contains internal file storage info
+            if data["storage_class"] == "L":
+                data.pop("uri")
+
+            # [TODO] Remote and Fetch
+            elif data["storage_class"] == "F":
+                pass
+            elif data["storage_class"] == "R":
+                pass
+            else:
+                # Error?
```

Comment:
I would not raise an error, I would `.pop`. by default? Better safe that sorry?

Reply:
If we do not raise an error the record will get published with incomplete files:
1. I add a FETCH file
2. I click publish

Once the record has been published the files cannot be added (bucket is locked). This is a limitation from the implementation, but records cannot be published until their files have been downloaded.